toucan_connectors/rok/rok_connector.py

```diff
@@ -14,33 +17,77 @@ class RokDataSource(ToucanDataSource):
 
 class RokConnector(ToucanConnector):
     data_source_model: RokDataSource
-
     host: str
     username: str
-    password: str
+    password: str = None
+    secret: str = None
+    authentified_with_rok_token: bool = False
 
     def _retrieve_data(self, data_source: RokDataSource) -> pd.DataFrame:
 
-        endpoint = f'{self.host}/graphql?DatabaseName={data_source.database}'
+        endpoint = f'{self.host}/graphql'
+
+        if self.authentified_with_rok_token:
+            if not data_source.live_data:
+                raise InvalidAuthenticationMethodError(
+                    'Request with ROK token is not possible while not in live data mode. Change the connector configuration to live data'
+                )
+
+            if not self.secret:
+                raise NoROKSecretAvailableError('secret not defined')
+
+            auth_query = """
+            query Auth($database: String!, $token: String!)
+            {authenticateUsingJWT(database: $database, token: $token)}"""
+
+            payload = {
+                'database': data_source.database,
+                'username': self.username,
+                'iat': datetime.utcnow(),
+            }
+
+            encoded_payload = encode(payload, self.secret, algorithm='HS256')
+            auth_vars = {
+                'database': data_source.database,
+                'jwt_token': encoded_payload.decode('utf-8'),
+            }
+            auth_res = requests.post(
+                endpoint, json={'query': auth_query, 'variables': auth_vars}
+            ).json()
+            rok_token = auth_res['data']['token']
 
-        auth_query = """
+        else:
+            endpoint = f'{endpoint}?DatabaseName={data_source.database}'
+
+            auth_query = """
             query Auth($database: String!, $user: String!, $password: String!)
             {authenticate(database: $database, user: $user, password: $password)}"""
-        auth_vars = {
-            'database': data_source.database,
-            'user': self.username,
-            'password': self.password,
-        }
-        auth_res = requests.post(
-            endpoint, json={'query': auth_query, 'variables': auth_vars}
-        ).json()
-        if 'errors' in auth_res:
-            raise ValueError(str(auth_res['errors']))
+            auth_vars = {
+                'database': data_source.database,
+                'user': self.username,
+                'password': self.password,
+            }
+            auth_res = requests.post(
+                endpoint, json={'query': auth_query, 'variables': auth_vars}
+            ).json()
+
+            if 'errors' in auth_res:
+                raise ValueError(str(auth_res['errors']))
+            rok_token = auth_res['data']['authenticate']
```

Comment:
For now, this is not too big, but it would be nice to extract this into an retrieve_token method of the connector, si it can be easily mocked, and keep the logici in retrieve_data easily readable